lib/output.js

```diff
@@ -1283,7 +1284,12 @@ function OutputStream(options) {
 
     // self should be AST_New.  decide if we want to show parens or not.
     function no_constructor_parens(self, output) {
-        return self.args.length == 0 && !output.option("beautify");
+        return self.args.length == 0 && !output.option("beautify") ||
+            !(self.expression instanceof AST_SymbolRef ||
+              self.expression instanceof AST_Call ||
+              self.expression instanceof AST_Function ||
+              self.expression instanceof AST_Assign
+            );
```

Comment:
None of these new conditions are exercised by your tests. Could you either add test support for them or drop this part of the patch?